cmake/CMakeLists.txt

```diff
@@ -441,30 +441,34 @@ if(HDF5_SUPPORT)
 
     if (HDF5_LIBRARY_DIR)
     	# result of the HDF5 ctest package
-    	
+
+
         if (WIN32)
-        	set(zlib_post lib)
+            # Windows
+            set(zlib_post zlib)
         	set(lib_ext lib)
+        elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            # macOS
+            set(zlib_post libz)
+            set(lib_ext dylib)
         else()
-        	set(lib_ext a)
```

Comment:
Hm, I now also see this change, this also needs to be conditional. Because we can't have the static builds reference dynamic libraries.

Reply:
> but do we need to add which paths it should look for zlib at or is it complete the way you suggested it

The `PATHS` should then be followed by the directories where to search.

I tend to add `NO_DEFAULT_PATH` so that cmake doens't get clever looking in other places.

For now, hack I think is fine, the cmake script needs a lot of work anyway.